### PR TITLE
Upstream merge

### PR DESCRIPTION
**Pull request by Carbonhell**
Do not merge.

### CODE REVIEW

**Comment by Amelia0010** [comment]
oh boy, he dun did it

**Comment by Carbonhell** [comment]
don't merge yet eh

**Comment by Amelia0010** [comment]
upstream merge = bad
cherrypick 178 commits = good

**Comment by JohnGinnane** [comment]
\>178 commits
isn't that more work though?

**Comment by Amelia0010** [comment]
more pain, more gain :^)

File: SQL/database_changelog.txt [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 or
-INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (4, 5);
+INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (4, 6);
 
 In any query remember to add a prefix to the table names if you use one.
 
 ----------------------------------------------------
 
+Version 4.6, 11 August 2018, by Jordie0608
+Modified table `messages`, adding column `expire_timestamp` to allow for auto-"deleting" messages.
```
Comment (JohnGinnane) [inline SQL/database_changelog.txt:14]: Carbon I think we need to run this SQL as server is expecting 4.6 but I think we're still on 4.4 from the ckey change

**Pull request merged**